public/app/features/dashboard/components/GenAI/utils.ts

```diff
@@ -121,3 +119,29 @@ export function getDashboardPanelPrompt(dashboard: DashboardModel): string {
   // So it is possibly that if we can condense it further it would be better
   return panelPrompt;
 }
+
+export function getFilteredPanelString(panel: PanelModel): string {
```

Comment:
we could probably incorporate this in the dashboard title / description generation as well as that also takes in panel info (currently limited to first 10 panels with descriptions) - can be a follow-up though

Reply:
Dashboard generation I think presently only gets titles and descriptions, not anything else. Only panel was sending the entire panel json.